internal/collector/postgres_metrics.go

```diff
@@ -110,3 +196,53 @@ func appendToJSONArray(a1, a2 json.RawMessage) (json.RawMessage, error) {
 
 	return merged, nil
 }
+
+func removeMetricsFromQueries(metricsToRemove []string,
+	queryMetricsArr []map[string]any,
+) []map[string]any {
+	// Iterate over the metrics that should be removed
+Outer:
+	for _, metricToRemove := range metricsToRemove {
+		// Iterate over array of query/metrics maps
+		for j, queryAndMetrics := range queryMetricsArr {
+			// Assert that the metrics key in the query/metrics map holds an array
+			if metricsArr, ok := queryAndMetrics["metrics"].([]any); ok {
+				// Iterate over the metrics array
+				for k, metric := range metricsArr {
+					// Assert that element in the metrics array is a map[string]any
+					if metricMap, ok := metric.(map[string]any); ok {
+						// Check to see if the metric_name matches the metricToRemove
+						if metricMap["metric_name"] == metricToRemove {
+							// Remove the metric. Since there won't ever be any
+							// duplicates, we will be exiting this loop early and
+							// therefore don't care about the order of the metrics
+							// array.
+							metricsArr[k] = metricsArr[len(metricsArr)-1]
+							metricsArr = metricsArr[:len(metricsArr)-1]
+							queryAndMetrics["metrics"] = metricsArr
+
+							// If the metrics array is empty, remove the query/metrics
+							// map entirely. Again, we don't care about order.
+							if len(metricsArr) == 0 {
+								queryMetricsArr[j] = queryMetricsArr[len(queryMetricsArr)-1]
+								queryMetricsArr = queryMetricsArr[:len(queryMetricsArr)-1]
+							}
+
+							// We found and deleted the metric, so we can continue
+							// to the next iteration of the Outer loop.
+							continue Outer
+						}
+					} else {
+						// TODO: What should we do if the assertion fails?
+						fmt.Printf("%s is not a map[string]any", metric)
```

Comment:
If this fails here, then something is wrong with (a) the file that we put in to the operator or (b) part of our marshalling/unmarshalling that wasn't caught when we did it -- is there any other way we could get an error at this point?

If (a), then our metrics collection will probably error out (and what does that look like?).

None of which answers your question, though! I just mean to say: if we've reached this point -- which seems unlikely -- we might have bigger problems. For now, I like the idea of logging the unexpected result but letting the process run on -- and we can revisit this in the future.